psiminer-core/src/main/kotlin/psi/language/KotlinHandler.kt

```diff
@@ -16,10 +17,25 @@ class KotlinHandler : LanguageHandler() {
 
     override val transformationType = KotlinTreeTransformation::class.java
     override val classPsiType = KtClass::class.java
-    override val methodPsiType = KtNamedFunction::class.java
+    override val methodPsiType = KtFunction::class.java
 
-    override fun collectFunctionCallsIdentifiers(root: PsiElement): List<PsiElement> =
-        PsiTreeUtil
-            .collectElements(root) { it.elementType.toString() == "IDENTIFIER" }
-            .filter { it.parent is KtReferenceExpression }
+    private val KtExpression.resolveToElement: PsiElement?
+        get() {
+            return when (this) {
+                is KtCallExpression -> referenceExpression()?.resolve()
+                is KtReferenceExpression -> this.resolve()
+                else -> null
+            }
+        }
+
+    override fun actionOnRecursiveCall(root: PsiElement, action: (PsiElement) -> Unit) {
+        PsiTreeUtil.collectElements(root) { it is KtCallExpression }
+            .filter {
+                val resolved = (it as KtCallExpression).resolveToElement
+                PsiManager.getInstance(root.project).areElementsEquivalent(root, resolved)
+            }
+            .forEach { callExpr ->
+                PsiTreeUtil.collectElements(callExpr) { it.elementType.toString() == "IDENTIFIER" }.forEach(action)
```

Comment:
Here you apply the action only to the identifiers, which is unclear from the function name.